topdown/builtins.go

```diff
@@ -54,6 +54,7 @@ type (
 		DistributedTracingOpts tracing.Options       // options to be used by distributed tracing.
 		rand                   *rand.Rand            // randomization source for non-security-sensitive operations
 		Capabilities           *ast.Capabilities
+		e                      *eval
```

Comment:
We need to give the builtins some means to evaluate stuff -- this, `*eval`, is _too much_. Good interfaces still need to be figured out, but there are two attempts below, `Resolve(ast.Ref)` and `ResolveTerm(*ast.Term)` 👇